ff/src/fields/models/cubic_extension.rs

```diff
@@ -128,11 +128,17 @@ impl<P: CubicExtParameters> One for CubicExtField<P> {
 
 impl<P: CubicExtParameters> Field for CubicExtField<P> {
     type BasePrimeField = P::BasePrimeField;
+    type BasePrimeFieldArr = Vec<Self::BasePrimeField>;
```

Comment:
We'll have to explicitly specify the type of the iterator; the one I'm suggesting below might not be the correct type, but is probably not too far from the correct one.
```suggestion

type BaseFieldIter<P> = <P::BaseField as Field>::BasePrimeFieldIter;

impl<P: CubicExtParameters> Field for CubicExtField<P> {
    type BasePrimeField = P::BasePrimeField;
    type BasePrimeFieldArr = Chain<BaseFieldIter<P>, Chain<BaseFieldIter<P>, BaseFieldIter<P>>>;
```